pkg/resourcemanager/webhook/nodeagentauthorizer/authorizer.go

```diff
@@ -0,0 +1,234 @@
+	}
+
+	if attrs.IsResourceRequest() {
+		requestResource := schema.GroupResource{Group: attrs.GetAPIGroup(), Resource: attrs.GetResource()}
+		switch requestResource {
+		case certificateSigningRequestResource:
+			return a.authorizeCertificateSigningRequest(ctx, requestLog, attrs)
+		case eventCoreResource, eventResource:
+			return a.authorizeEvent(requestLog, attrs)
+		case leaseResource:
+			return a.authorizeLease(ctx, requestLog, machineName, attrs)
+		case nodeResource:
+			return a.authorizeNode(ctx, requestLog, machineName, attrs)
+		case secretsResource:
+			return a.authorizeSecret(ctx, requestLog, machineName, attrs)
+		}
+	}
+
+	return auth.DecisionNoOpinion, "", nil
+}
+
+func (a *authorizer) authorizeCertificateSigningRequest(ctx context.Context, log logr.Logger, attrs auth.Attributes) (auth.Decision, string, error) {
+	if ok, reason := a.checkSubresource(log, attrs); !ok {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	if allowed, reason := a.checkVerb(log, attrs, "get", "create"); !allowed {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	if attrs.GetVerb() == "create" {
+		return auth.DecisionAllow, "", nil
+	}
+
+	csr := &certificatesv1.CertificateSigningRequest{}
+	if err := a.targetClient.Get(ctx, types.NamespacedName{Name: attrs.GetName()}, csr); err != nil {
+		return auth.DecisionNoOpinion, "", err
+	}
+
+	if csr.Spec.Username != attrs.GetUser().GetName() {
+		log.Info("Denying authorization because the CSR is for a different user", "csrUsername", csr.Spec.Username)
+		return auth.DecisionNoOpinion, "gardener-node-agent is only allowed to get CSRs for its own user", nil
+	}
+
+	return auth.DecisionAllow, "", nil
+}
+
+func (a *authorizer) authorizeEvent(log logr.Logger, attrs auth.Attributes) (auth.Decision, string, error) {
+	if ok, reason := a.checkSubresource(log, attrs); !ok {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	if allowed, reason := a.checkVerb(log, attrs, "create", "patch"); !allowed {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	return auth.DecisionAllow, "", nil
+}
+
+func (a *authorizer) authorizeLease(ctx context.Context, log logr.Logger, machineName string, attrs auth.Attributes) (auth.Decision, string, error) {
+	if ok, reason := a.checkSubresource(log, attrs); !ok {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	allowedVerbs := []string{"get", "list", "watch", "create", "update"}
+	if allowed, reason := a.checkVerb(log, attrs, allowedVerbs...); !allowed {
+		return auth.DecisionNoOpinion, reason, nil
+	}
+
+	machine := &machinev1alpha1.Machine{}
+	if err := a.sourceClient.Get(ctx, client.ObjectKey{Name: machineName, Namespace: a.machineNamespace}, machine); err != nil {
+		return auth.DecisionNoOpinion, "", err
+	}
+
+	node := machine.Labels[machinev1alpha1.NodeLabelKey]
+	if node == "" {
+		log.Info("Denying authorization for machine to the lease because the machine does not have a \"node\" label", "machine", machineName)
```

Comment:
```suggestion
		log.Info(`Denying request because the machine does not have a "node" label`, "machineName", machineName)
```

see logging guidelines